solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -298,37 +303,59 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     // Pick the action
     SolrParams params = req.getParams();
     String a = params.get(CoreAdminParams.ACTION);
-    if (a != null) {
-      CollectionAction action = CollectionAction.get(a);
-      if (action == null) {
-        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "Unknown action: " + a);
-      }
-      final String collection = params.get(COLLECTION);
-      MDCLoggingContext.setCollection(collection);
-      TraceUtils.setDbInstance(req, collection);
-      CollectionOperation operation = CollectionOperation.get(action);
-      if (log.isDebugEnabled()) {
-        log.debug(
-            "Invoked Collection Action: {} with params {}", action.toLower(), req.getParamString());
-      }
-      invokeAction(req, rsp, cores, action, operation);
-    } else {
+    if (a == null) {
```

Comment:
I don't know how much celan up you want....   but `a` as a variable?   Makes me think there must be a `b` somewhere around ;-).   `actionParam`?  `action`?

Reply:
Agreed.  IMO it comes back to your first comment though: "I don't know how much clean up you want".

Since we're aiming to make similar changes across all of Solr's 200+ APIs, I'm leery of straying into any sort of cleanup that's not strictly necessary for our API goals - even though I totally agree that it's absolutely needed in general.

Reply:
Yeah...   You are right...  sigh...   Good for you for staying on point and not letting me distract you!   (This is why my PR's tend to grow over time :-(  ).    

Any thoughts on how we could track these for future fixing?   I worry about creating a million JIRA's...    Notes in the spreadsheet "come back and check this"?